### PR TITLE
feat: dump operation state machine log

### LINKED ISSUE #5342: State log inspector
`fedimint-cli` should have a dev function that lets us inspect the state machine log.

### PR DESCRIPTION
**Pull request by elsirion**
Allows debugging state machine runs:

```
$ fedimint-cli ln-invoice --amount 42
2024-05-24T15:45:31.581799Z  WARN fedimint_cli::client: Command deprecated. Use `fedimint-cli module ln invoice` instead.
{
  "operation_id": "168081e9aecf41810939c2edfa3f96b0dd033b9c400ef82744d3172577bd403e",
  "invoice": "lnbcrt420p1pn9pdsmdqqpp5z6qgr6dweaqczzfectkl50ukkrwsxwuugq80sf6y6vtj2aaagqlqsp5stlsv0w70qps7uuhvtsarrmkgzvvldgl5a39rfhfellqah0fpmrs9qrsgqcqpjnp4qdu7qew9u9zxjchtsryrhttgmmh4wh4qa8utzf4hs7k480sz0q457xqyz5vqrzjqv4j30k3sqvykdc78pu87jy0dv6g4y7r4wtywhhn3z9xy4nh5awssqqqqqqqqqqqqyqqqqqqqqqqqqqqrcr9yqvd0wec2kv8tagz90g39mp8cgvfzulte2uvsfwgyhuf2k9gd3c95zqqqwqqqqqgqqyqqqqlgqqqqqqgq2qpjk2976xqpsjehrcu8sl6g3a4nfz5ncw4ev3677wyg5cjkw7n46zqqqqqqqqqqqqqsqqqqqqqqqqqqqq0qpyuheyt2yecmy4jwurmdjnjneh2n2c5ff0q6g9f8xfd6wtcydz5rv49fudwsd9hgrlc969ly5agpgn3rk7dpmwdu5ukcj3qs6xdg06qp8k33w4"
}

$ fedimint-cli dev list-operation-states 168081e9aecf41810939c2edfa3f96b0dd033b9c400ef82744d3172577bd403e
{
  "states": [
    {
      "active": false,
      "module_instance": 0,
      "creation_time": "2024-05-24T15:45:31.588438019",
      "end_time": "2024-05-24T15:45:31.783662035",
      "state": "Receive(LightningReceiveStateMachine { operation_id: OperationId(168081e9_77bd403e), state: SubmittedOffer(LightningReceiveSubmittedOffer { offer_txid: 0x91a4b38b65fac49020e3cfcb830214a5a3011f157c70741463d781f944a2b8bc, invoice: Bolt11Invoice { signed_invoice: SignedRawBolt11Invoice { raw_invoice: RawBolt11Invoice { hrp: RawHrp { currency: Regtest, raw_amount: Some(420), si_prefix: Some(Pico) }, data: RawDataPart { timestamp: PositiveTimestamp(1716565531s), tagged_fields: [KnownSemantics(Description(Description(UntrustedString(\"\")))), KnownSemantics(PaymentHash(Sha256(0x168081e9aecf41810939c2edfa3f96b0dd033b9c400ef82744d3172577bd403e))), KnownSemantics(PaymentSecret(PaymentSecret([130, 255, 6, 61, 222, 120, 3, 15, 115, 151, 98, 225, 209, 143, 118, 64, 152, 207, 181, 31, 167, 98, 81, 166, 233, 207, 254, 14, 221, 233, 14, 199]))), KnownSemantics(Features([0, 65])), KnownSemantics(MinFinalCltvExpiryDelta(MinFinalCltvExpiryDelta(18))), KnownSemantics(PayeePubKey(PayeePubKey(PublicKey(4f2b7802be53ad87b726b1f8e9a05e57efde68ad3bc880eb626944e1c565e0793dd522abdc1f8f7c890110c53e49c53c186b80fe3f7dab067c07aa95d34ee539)))), KnownSemantics(ExpiryTime(ExpiryTime(86400s))), KnownSemantics(PrivateRoute(PrivateRoute(RouteHint([RouteHintHop { src_node_id: PublicKey(085da77756628a88f35e4796abc3938a346b8f487f78381e374b1880d1be282be9abd81978b48bc8d22cc53c6ab5d3c7f23646803f669da043d732bc510fedce), short_channel_id: 1, fees: RoutingFees { base_msat: 0, proportional_millionths: 0 }, cltv_expiry_delta: 30, htlc_minimum_msat: None, htlc_maximum_msat: None }])))), KnownSemantics(PrivateRoute(PrivateRoute(RouteHint([RouteHintHop { src_node_id: PublicKey(410b8e0d15ab12bf04b9041957797d2e1243f8845d227a45a0be0eb30a67f71a9bc93ecb6997f94d8cde1d79e45cc80696d0b3d16963ab0bb4944e558f1db4ea), short_channel_id: 123145302376449, fees: RoutingFees { base_msat: 1000, proportional_millionths: 1 }, cltv_expiry_delta: 80, htlc_minimum_msat: None, htlc_maximum_msat: None }, RouteHintHop { src_node_id: PublicKey(085da77756628a88f35e4796abc3938a346b8f487f78381e374b1880d1be282be9abd81978b48bc8d22cc53c6ab5d3c7f23646803f669da043d732bc510fedce), short_channel_id: 1, fees: RoutingFees { base_msat: 0, proportional_millionths: 0 }, cltv_expiry_delta: 30, htlc_minimum_msat: None, htlc_maximum_msat: None }]))))] } }, hash: [66, 124, 226, 164, 26, 180, 151, 29, 214, 237, 188, 48, 25, 113, 186, 109, 177, 95, 126, 218, 29, 127, 194, 164, 177, 235, 128, 64, 168, 28, 80, 6], signature: Bolt11InvoiceSignature(RecoverableSignature(09397c916a2671b2564ee0f6d94e53cdd53562894bc1a41527325ba72f0468a83654a9e35d0696e81ff05d17e4a750144e23b79a1db9bca72d894410d19a87e801)) } }, receiving_key: Personal(KeyPair(#3461bef422aa996b)) }) })"
    },
    {
      "active": false,
      "module_instance": 65535,
      "creation_time": "2024-05-24T15:45:31.588654008",
      "end_time": "2024-05-24T15:45:31.782620263",
      "state": "OperationState { operation_id: OperationId(168081e9_77bd403e), state: Created(Transaction { inputs: [], outputs: [V0(Offer(IncomingContractOffer { amount: Amount { msats: 42 }, hash: 0x168081e9aecf41810939c2edfa3f96b0dd033b9c400ef82744d3172577bd403e, encrypted_preimage: EncryptedPreimage(Ciphertext(G1Projective { x: 0x057f3c1687d626d5a0d14586cbeea13a6a6ee6ae2ee4d8e5e36d5d7c9a7e2fb8c323378e0681f034c7c5d5ddd44c320e, y: 0x0611b6b309ee22732aefbb1dd6272b01b1fec54874ddf2b9ee9ef3c0b99b4cd70c54e501b84bd1b521848b2f77c85963, z: 0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001 }, [107, 188, 191, 253, 109, 11, 68, 95, 110, 155, 0, 67, 215, 32, 180, 238, 116, 57, 26, 160, 251, 43, 65, 221, 220, 215, 23, 18, 24, 62, 124, 141, 252], G2Projective { x: 0x0fa45443ab46b4f884f1b68be0110dbc3d677673f7bf1f7cba3a8f001ec7e38fcc98bcf25b94de46c798a999d53e5626 + 0x15f09558e580c49a657c5a4842d83601c5e22ec66db44091611b6663afb36af750f9eca8a4ff2215495ecb4aba074841*u, y: 0x0a18e38a37d981c10546a1b5255d5a1aaa7cb12bbb093277f262ced088e65e5898e263137446a289293aedd6569445d5 + 0x00af49b4d8926a896d2e9045277cb44550dc7dae34f2326ca348f90344c6059387d9a28a769c54a80bc82b8e27848f81*u, z: 0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001 + 0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000*u })), expiry_time: None }))], nonce: [170, 213, 166, 41, 39, 79, 47, 217], signatures: NaiveMultisig([]) }) }"
    },
    {
      "active": false,
      "module_instance": 65535,
      "creation_time": "2024-05-24T15:45:31.782792880",
      "end_time": "2024-05-24T15:45:31.782792920",
      "state": "OperationState { operation_id: OperationId(168081e9_77bd403e), state: Accepted(0x91a4b38b65fac49020e3cfcb830214a5a3011f157c70741463d781f944a2b8bc) }"
    },
    {
      "active": true,
      "module_instance": 0,
      "creation_time": "2024-05-24T15:45:31.783726417",
      "state": "Receive(LightningReceiveStateMachine { operation_id: OperationId(168081e9_77bd403e), state: ConfirmedInvoice(LightningReceiveConfirmedInvoice { invoice: Bolt11Invoice { signed_invoice: SignedRawBolt11Invoice { raw_invoice: RawBolt11Invoice { hrp: RawHrp { currency: Regtest, raw_amount: Some(420), si_prefix: Some(Pico) }, data: RawDataPart { timestamp: PositiveTimestamp(1716565531s), tagged_fields: [KnownSemantics(Description(Description(UntrustedString(\"\")))), KnownSemantics(PaymentHash(Sha256(0x168081e9aecf41810939c2edfa3f96b0dd033b9c400ef82744d3172577bd403e))), KnownSemantics(PaymentSecret(PaymentSecret([130, 255, 6, 61, 222, 120, 3, 15, 115, 151, 98, 225, 209, 143, 118, 64, 152, 207, 181, 31, 167, 98, 81, 166, 233, 207, 254, 14, 221, 233, 14, 199]))), KnownSemantics(Features([0, 65])), KnownSemantics(MinFinalCltvExpiryDelta(MinFinalCltvExpiryDelta(18))), KnownSemantics(PayeePubKey(PayeePubKey(PublicKey(4f2b7802be53ad87b726b1f8e9a05e57efde68ad3bc880eb626944e1c565e0793dd522abdc1f8f7c890110c53e49c53c186b80fe3f7dab067c07aa95d34ee539)))), KnownSemantics(ExpiryTime(ExpiryTime(86400s))), KnownSemantics(PrivateRoute(PrivateRoute(RouteHint([RouteHintHop { src_node_id: PublicKey(085da77756628a88f35e4796abc3938a346b8f487f78381e374b1880d1be282be9abd81978b48bc8d22cc53c6ab5d3c7f23646803f669da043d732bc510fedce), short_channel_id: 1, fees: RoutingFees { base_msat: 0, proportional_millionths: 0 }, cltv_expiry_delta: 30, htlc_minimum_msat: None, htlc_maximum_msat: None }])))), KnownSemantics(PrivateRoute(PrivateRoute(RouteHint([RouteHintHop { src_node_id: PublicKey(410b8e0d15ab12bf04b9041957797d2e1243f8845d227a45a0be0eb30a67f71a9bc93ecb6997f94d8cde1d79e45cc80696d0b3d16963ab0bb4944e558f1db4ea), short_channel_id: 123145302376449, fees: RoutingFees { base_msat: 1000, proportional_millionths: 1 }, cltv_expiry_delta: 80, htlc_minimum_msat: None, htlc_maximum_msat: None }, RouteHintHop { src_node_id: PublicKey(085da77756628a88f35e4796abc3938a346b8f487f78381e374b1880d1be282be9abd81978b48bc8d22cc53c6ab5d3c7f23646803f669da043d732bc510fedce), short_channel_id: 1, fees: RoutingFees { base_msat: 0, proportional_millionths: 0 }, cltv_expiry_delta: 30, htlc_minimum_msat: None, htlc_maximum_msat: None }]))))] } }, hash: [66, 124, 226, 164, 26, 180, 151, 29, 214, 237, 188, 48, 25, 113, 186, 109, 177, 95, 126, 218, 29, 127, 194, 164, 177, 235, 128, 64, 168, 28, 80, 6], signature: Bolt11InvoiceSignature(RecoverableSignature(09397c916a2671b2564ee0f6d94e53cdd53562894bc1a41527325ba72f0468a83654a9e35d0696e81ff05d17e4a750144e23b79a1db9bca72d894410d19a87e801)) } }, receiving_key: Personal(KeyPair(#3461bef422aa996b)) }) })"
    }
  ]
}
```

Fixes #5342

### CODE REVIEW

File: fedimint-cli/src/client.rs [hunk 1]
```diff
@@ -693,3 +726,21 @@ pub struct InfoResponse {
     total_num_notes: usize,
     denominations_msat: TieredCounts,
 }
+
+fn time_to_iso8601(time: &SystemTime) -> String {
+    const ISO8601_CONFIG: iso8601::EncodedConfig = iso8601::Config::DEFAULT
+        .set_formatted_components(iso8601::FormattedComponents::DateTime)
+        .set_time_precision(TimePrecision::Second {
+            decimal_digits: None,
```
Comment (elsirion) [inline fedimint-cli/src/client.rs:734]: Should have ms granularity

**Review by dpc — COMMENTED** [review]
Millis would be really useful, and maybe it should be dev command, but LGTM otherwise.

**Review by dpc — APPROVED** [review]

**Comment by dpc** [comment]
That MacOS flake extra severe recently. Hopefully it will get fixed sometime soon.

**Comment by fedimint-backports** [comment]
Backport failed for `releases/v0.3`, because it was unable to cherry-pick the commit(s).

Please cherry-pick the changes locally and resolve any conflicts.
```bash
git fetch origin releases/v0.3
git worktree add -d .worktree/backport-5359-to-releases/v0.3 origin/releases/v0.3
cd .worktree/backport-5359-to-releases/v0.3
git switch --create backport-5359-to-releases/v0.3
git cherry-pick -x e12d82c5236287ca67bb9c5fc7d3f5f3ec7c2405
```

**Pull request merged**